odl/space/base_ntuples.py

```diff
@@ -448,6 +437,66 @@ def is_cn(self):
         """
         return (not self._is_real) and self._is_floating
 
+    def as_complex_space(self, dtype=None):
```

Comment:
Same comments as above apply here. Dislike the possible explosion of functions, better have one unified.